lib/stat_tracker.rb

```diff
@@ -23,6 +23,49 @@ def initialize(files)
     end
   end
 
+  def all_season_game_id(season)
+    @games.map do |game|
+      game.game_id if game.season == season
+    end.compact 
+  end 
+
+  def winningest_coach(season)
+    coach_wins = @game_teams.each_with_object(Hash.new([0,0])) do |game, hash|
+      if all_season_game_id(season).include?(game.game_id) 
+        if game.result == "WIN"
+          hash[game.head_coach] = [1 + hash[game.head_coach][0], 1 + hash[game.head_coach][1]]
+        else
+          hash[game.head_coach] = [hash[game.head_coach][0], 1 + hash[game.head_coach][1]]
```

Comment:
Logic block for the each_with_object looks great. Setting and incrementing array blocks as values can be an easy way to get wrapped around.